### PR TITLE
Tidy

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

**Review by panquez — APPROVED** [review]

File: src/geode/basic/cell_array.cpp [hunk 1]
```diff
@@ -33,7 +33,7 @@ namespace geode
     {
     public:
         Impl() = default;
-        Impl( std::array< index_t, dimension > cells_number )
+        explicit Impl( std::array< index_t, dimension > cells_number )
```
Comment (BotellaA) [inline src/geode/basic/cell_array.cpp:36]: warning: constructor does not initialize these fields: cells_number_ [cppcoreguidelines-pro-type-member-init]
```cpp
ublic:
                                           ^
```

File: src/geode/basic/filename.cpp [hunk 2]
```diff
@@ -29,13 +29,13 @@ namespace geode
 {
     std::string filename_with_extension( absl::string_view path )
     {
-        ghc::filesystem::path filepath{ to_string( path ) };
+        const ghc::filesystem::path filepath{ to_string( path ) };
```
Comment (BotellaA) [inline src/geode/basic/filename.cpp:32]: warning: variable 'filepath' is not initialized [cppcoreguidelines-init-variables]

```suggestion
        const ghc::filesystem::path filepath = 0{ to_string( path ) };
```

File: src/geode/basic/filename.cpp [hunk 3]
```diff
@@ -29,13 +29,13 @@ namespace geode
...
         return filepath.filename().string();
     }
 
     std::string filename_without_extension( absl::string_view path )
     {
-        ghc::filesystem::path filepath{ to_string( path ) };
+        const ghc::filesystem::path filepath{ to_string( path ) };
```
Comment (BotellaA) [inline src/geode/basic/filename.cpp:38]: warning: variable 'filepath' is not initialized [cppcoreguidelines-init-variables]

```suggestion
        const ghc::filesystem::path filepath = 0{ to_string( path ) };
```

File: src/geode/basic/progress_logger.cpp [hunk 4]
```diff
@@ -33,18 +33,18 @@
 
 namespace
 {
-    static constexpr absl::Duration SLEEP = absl::Seconds( 1 );
-}
+    constexpr absl::Duration SLEEP = absl::Seconds( 1 );
```
Comment (BotellaA) [inline src/geode/basic/progress_logger.cpp:36]: warning: invalid case style for constexpr variable 'SLEEP' [readability-identifier-naming]

```suggestion
    constexpr absl::Duration sleep = absl::Seconds( 1 );
```

File: src/geode/basic/string.cpp [hunk 5]
```diff
@@ -35,37 +35,40 @@ namespace geode
...
+            "\" to index" );
         return value;
     }
 
     int string_to_int( absl::string_view string )
     {
-        int value;
-        const auto ok = absl::SimpleAtoi( string, &value );
-        OPENGEODE_EXCEPTION( ok, "[string_to_int] Cannot convert string \"",
+        int value{ 0 };
```
Comment (BotellaA) [inline src/geode/basic/string.cpp:48]: warning: variable 'value' of type 'int' can be declared 'const' [misc-const-correctness]

```suggestion
        int const value{ 0 };
```

File: src/geode/basic/string.cpp [hunk 6]
```diff
@@ -35,37 +35,40 @@ namespace geode
...
         return value;
     }
 
     float string_to_float( absl::string_view string )
     {
-        float value;
-        const auto ok = absl::SimpleAtof( string, &value );
-        OPENGEODE_EXCEPTION( ok, "[string_to_float] Cannot convert string \"",
-            string, "\" to float" );
+        float value{ 0 };
```
Comment (BotellaA) [inline src/geode/basic/string.cpp:57]: warning: variable 'value' of type 'float' can be declared 'const' [misc-const-correctness]

```suggestion
        float const value{ 0 };
```

**Review by BotellaA — COMMENTED** [review]
clang-tidy made some suggestions

**Review by MelchiorSchuh — APPROVED** [review]

**Pull request merged**